openapi_python_client/parser/openapi.py

```diff
@@ -260,12 +260,22 @@ class GeneratorData:
     enums: Dict[str, EnumProperty]
 
     @staticmethod
-    def from_dict(d: Dict[str, Dict[str, Any]]) -> Union["GeneratorData", GeneratorError]:
+    def from_dict(d: Dict[str, Any]) -> Union["GeneratorData", GeneratorError]:
         """ Create an OpenAPI from dict """
         try:
             openapi = oai.OpenAPI.parse_obj(d)
         except ValidationError as e:
-            return GeneratorError(header="Failed to parse OpenAPI document", detail=str(e))
+            detail = str(e)
+            if "swagger" in d:
+                detail = (
+                    "You may be trying to use a Swagger document; this is not supported by this project.\n\n" + detail
+                )
```

Comment:
Why not just check for this or the `openapi` field before attempting to parse the dict?

Reply:
1. I don't think it's illegal to have extra fields (e.g. `swagger`) so I don't want to disallow that if the spec is otherwise valid
2. Validating the format of `openapi` seems like a task for Pydantic since the semver requirement is part of the OpenAPI spec.